File: app/lib/services/s3_client_app.dart
/// CSTCloud Data Capsule client-app bindings (must match AccessKey creation in console).
class S3ClientAppOption {
  final String id;
  final String label;

  const S3ClientAppOption({required this.id, required this.label});
}

const s3ClientAppOptions = [
  S3ClientAppOption(id: 's3drive', label: 'S3Drive'),
  S3ClientAppOption(id: 's3browser', label: 'S3Browser'),
  S3ClientAppOption(id: 'rclone', label: 'Rclone'),
  S3ClientAppOption(id: 'obsidian', label: 'Obsidian'),
  S3ClientAppOption(id: 'cherry_studio', label: 'Cherry Studio'),
];

bool s3EndpointRequiresClientApp(String endpoint) {
  return endpoint.toLowerCase().contains('cstcloud.cn');
}

File: web/src/lib/s3ClientApp.ts
export const S3_CLIENT_APP_OPTIONS = [
  { id: 's3drive', label: 'S3Drive' },
  { id: 's3browser', label: 'S3Browser' },
  { id: 'rclone', label: 'Rclone' },
  { id: 'obsidian', label: 'Obsidian' },
  { id: 'cherry_studio', label: 'Cherry Studio' },
] as const;

export type S3ClientAppId = (typeof S3_CLIENT_APP_OPTIONS)[number]['id'];

export function s3EndpointRequiresClientApp(endpoint: string): boolean {
  return endpoint.toLowerCase().includes('cstcloud.cn');
}

export const S3_DEFAULT_USER_AGENT = 'ShrimpSend/1.0 S3Compat';
